android/src/main/java/com/swmansion/rnscreens/ScreenModalFragment.kt

```diff
@@ -0,0 +1,298 @@
+ import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+ import com.swmansion.rnscreens.bottomsheet.BottomSheetDialogRootView
+ import com.swmansion.rnscreens.bottomsheet.BottomSheetDialogScreen
+ import com.swmansion.rnscreens.events.ScreenDismissedEvent
+ import com.swmansion.rnscreens.ext.parentAsView
+ import com.swmansion.rnscreens.ext.recycle
+ import kotlin.math.max
+
+class ScreenModalFragment : BottomSheetDialogFragment, ScreenStackFragmentWrapper {
+    override lateinit var screen: Screen
+
+    // Nested containers
+    override val childScreenContainers = ArrayList<ScreenContainer>()
+
+    private val container: ScreenStack?
+        get() = screen.container as? ScreenStack
+
+    /**
+     * Dialog instance. Note that we are responsible for creating the dialog.
+     * This member is valid after `onCreateDialog` method runs.
+     */
+    private lateinit var sheetDialog: BottomSheetDialog
+
+    /**
+     * Behaviour attached to bottom sheet dialog.
+     * This member is valid after `onCreateDialog` method runs.
+     */
+    private val behavior
+        get() = sheetDialog.behavior
+
+    override val fragment: Fragment
+        get() = this
+
+    constructor() {
+        throw IllegalStateException(
+            "Screen fragments should never be restored. Follow instructions from https://github.com/software-mansion/react-native-screens/issues/17#issuecomment-424704067 to properly configure your main activity."
+        )
+    }
+
+    constructor(screen: Screen) : super() {
+        this.screen = screen
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        // Right now whole purpose of this Fragment is to be displayed as a dialog.
+        // I've experimented with setting false here, but could not get it to work.
+        showsDialog = true
+    }
+
+     // We override this method to provide our custom dialog type instead of the default Dialog.
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        configureDialogAndBehaviour()
+
+        val rootView = BottomSheetDialogRootView(screen.reactContext, screen.reactEventDispatcher!!)
+
+//        if (screen.isSheetGrabberVisible) {
+//            val dragHandle = BottomSheetDragHandleView(requireContext()).apply {
+//                layoutParams = ViewGroup.LayoutParams(
+//                    ViewGroup.LayoutParams.MATCH_PARENT,
+//                    ViewGroup.LayoutParams.WRAP_CONTENT,
+//                )
+//            }
+//            rootView.addView(dragHandle)
+//        }
+
+        rootView.addView(screen.recycle())
+        sheetDialog.setContentView(rootView)
+
+        rootView.parentAsView()?.clipToOutline = true
+
+        return sheetDialog
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? = null
```

Comment:
The thing is I don't remember exactly right now... Damn. I recall that creating a view here, instead of using `setContentView` on dialog directly would trigger different code path and different lifecycle methods to be called. It was important that the `onCreateView` didn't return a non-null object here.